observability-lib/grafana/dashboard.go

```diff
@@ -103,9 +104,11 @@ func (db *Dashboard) DeployToGrafana(options *DeployOptions) error {
 			alert.FolderUID = folder.UID
 			alert.Annotations["__dashboardUid__"] = *newDashboard.UID
 
-			panelId := panelIDByTitle(db.Dashboard, alert.Title)
+			panelId := panelIDByTitle(db.Dashboard, alert.Annotations["panel_title"])
+			// we can clean it up as it was only used to get the panelId
+			delete(alert.Annotations, "panel_title")
 			if panelId != "" {
-				alert.Annotations["__panelId__"] = panelIDByTitle(db.Dashboard, alert.Title)
+				alert.Annotations["__panelId__"] = panelId
```

Comment:
why this `__` prefix + suffix combo?

Reply:
its internal grafana API that requires this